modules/javafx.controls/src/main/java/javafx/scene/control/cell/ChoiceBoxTreeCell.java

```diff
@@ -294,7 +294,8 @@ public final void setConverter(StringConverter<T> value) {
 
     /** {@inheritDoc} */
     @Override public void startEdit() {
-        if (! isEditable() || ! getTreeView().isEditable()) {
+        super.startEdit();
+        if (!isEditing()) {
             return;
         }
 
```

Comment:
(darn, can't add the important lines - which is backing out if treeItem is null)

The re-ordering leads to change of behavior, here's a test that's passing/failing before/after:

    /**
     * change of behavior: cell must not be editing if treeItem == null.
     * fails with fix, passes without
     */
    @Test
    public void testChoiceBoxTreeCellEditing() {
        TreeView<String> treeView = new TreeView<>();
        treeView.setEditable(true);
        ChoiceBoxTreeCell<String> cell = new ChoiceBoxTreeCell<>();
        cell.updateTreeView(treeView);
        cell.updateItem("TEST", false);
        
        cell.startEdit();
        assertFalse(cell.isEditing());
        assertNull(cell.getGraphic());
    }
    
same for ComboBoxTreeCell

Reply:
Hm.. weird that the super class is firing an edit event even with `treeItem = null`. Maybe this should be investigated in a follow-up. Good catch though. :)

Reply:
yeah, there are whacky aspects:

- we can switch a cell that's not attached to its surroundings (like table, column, treeItem, off-range) into editing state - the only constraint being that it's not empty
- the other way round: a cell that didn't switch into editing will nevertheless fire a editStart on its target

Definitely leeway for improvements ;)

Reply:
> the other way round: a cell that didn't switch into editing will nevertheless fire a editStart on its target

FYI: filed [JDK-8274433](https://bugs.openjdk.java.net/browse/JDK-8274433)